src/spetlr/etl/extractors/check_schema_extractor.py

```diff
@@ -0,0 +1,59 @@
+from typing import List, Protocol
+
+import pyspark.sql.types as T
+from pyspark.sql import DataFrame
+
+from spetlr.etl import Extractor
+
+
+class Readable(Protocol):
+    def read(self) -> DataFrame:
+        pass
+
+
+class CheckSchemaExtractor(Extractor):
+    """This extractor will extract from any object that has a .read() method.
+    Subsequently, it will test the read dataframe against a schema stored in the object.
+    Specifically, the schemas are compared for breaking changes.
+    Breaking changes include:
+    - Adding a column
+    - Removing a column
+    - Changing a column name
+    """
+
+    def __init__(self, handle: Readable, dataset_key: str):
+        super().__init__(dataset_key=dataset_key)
+        self.handle = handle
+
+    def read(self) -> DataFrame:
+        df = self.handle.read()
+        df_schema = df.schema
+        expected_schema = self.handle.schema
+
+        schema_first_names = self._schemaColNames(df_schema)
+        schema_second_names = self._schemaColNames(expected_schema)
+
+        sorted_schema_first_names = sorted(schema_first_names)
+        sorted_schema_second_names = sorted(schema_second_names)
+
+        if len(schema_first_names) != len(schema_second_names):
+            raise Exception(
+                f"""Schemas have different number of columns.
+                First schema:
+                {schema_first_names}
+                Second schema:
+                {schema_second_names}"""
+            )
+
+        for i in range(len(sorted_schema_first_names)):
+            if sorted_schema_first_names[i] != sorted_schema_second_names[i]:
+                raise Exception(
+                    f"""Schemas are not equal!
+                    First schema:
+                    {schema_first_names}
+                    Second schema:
+                    {schema_second_names}"""
+                )
+
+    def _schemaColNames(self, schema: T.StructType) -> List[str]:
```

Comment:
It came to my mind - that we might also need to make it possible to check whether the data types are as expected?

Let's again say that it is a .csv file - we might end up extracting wrong data (or no data) if the datatypes cannot be read correctly.